packages/next-on-pages/src/buildApplication/buildMetadataFiles.ts

```diff
@@ -41,15 +44,23 @@ ${Object.entries(nextStaticHeaders)
 	}
 }
 
-async function buildRoutes(outputDir: string) {
+async function buildRoutes(
+	outputDir: string,
+	{ staticAssets }: BuildMetadataFilesOpts,
+) {
+	const nextStaticRegex = /^(.*\/_next\/static)\/.+$/;
+	const nextStaticAsset = staticAssets.find(a => nextStaticRegex.test(a));
+	const nextStaticPath =
+		nextStaticAsset?.match(nextStaticRegex)?.[1] ?? '/_next/static';
```

Comment:
Nit:
This is sort of clear to me but I feel like we could make the code more easy to read if we extracted this in a properly commented function, something like:
![Screenshot 2023-07-24 at 13 32 58](https://github.com/cloudflare/next-on-pages/assets/61631103/3c65b813-5ba2-49e3-a3f8-f1f62b27d71a)

But if you dislike extracting it this way (maybe it is unnecessary/unneeded?) I am also fine keeping the code as it, I leave it up to you 🙂